doc/man1/openssl-smime.pod.in

```diff
@@ -177,7 +178,7 @@ If not specified triple DES is used. Only used with B<-encrypt>.
 
 When verifying a message normally certificates (if any) included in
 the message are searched for the signing certificate. With this option
-only the certificates specified in the B<-certfile> option are used.
+only the certificates specified in the B<-untrusted> option are used.
```

Comment:
This paragraph is IMNSHO very confusing both **before** and **after** the proposed change. Poor grammar aside (what is "verifying a message normally certificates", where does the comma go?):

The text seems to want to say that

1. The immediate (EE) signer of the message is normally obtained from the certificates included in the message.
2. With the new option the EE certificate is instead obtained from the (till now) `-certfile` file.
3. It then seems to say that once the EE certificate is found, some set of certificates is still used as untrusted certificates to build the chain, but rather unclear which!

The key issue is the muddy meaning of *supplied* in "3". Is that the certificates in the message, or the the `-certfile`?  What was the original behaviour?

Finally, I think it is confusing to call this `-untrusted`, when signing. In that context `-certfile` is more clear.

My suggestion is to NOT adopt this PR, until we first clarify the documentation with no other changes, and then consider whether a new option name is warranted.

Reply:
Before this PR, it seems that using `-certfile` always ignored any certificates in the message, using only `-certfile` for both the EE cert and chain construction. This could potentially work around outdated issuers in the message.

Now the message has the union of the original and `-certfile` certificates. Is this behaviour change deliberate and desirable? Now also, the EE certificate (again possibly expired in the message) will be taken from the message, rather than `-certfile`.

Reply:
To me, the intended meaning of "The supplied certificates can still be used as untrusted CAs however." is clear: the "supplied certificates" refer to those given with the CLI option (not those in the message).

I agree that the grammar should be improved by adding a comma in the first sentence, 
and the last sentence of this section could be re-phrased to avoid misunderstandings.

This PR so far does not intend to improve the documentation, but this could be added.

Reply:
Unfortunately, the function `PKCS7_add_certificate()` used here is not documented.
I find it even more unfortunate and confusing that the certs that can be added this way play a double role:
they are used for both signer (EE) certs and for untrusted (intermediate) CA certs.

This PR focused on the 'untrusted' aspect, but as @vdukhovni pointed out above, there is also the 'signer' aspect.

I would have been much better if the two aspects had been kept apart, and then it would be straightforward to have both a  `-signer_certs` and an `-untrusted` option rather than the ambivalent `-certfile` option.

I meanwhile agree with Viktor that, given that weird semantics, the old name `-certfile` was better.

Reply:
So I'm going to revert adding the alias `-untrusted`.
I will also add improvements to the documentation.

Reply:
But the P.R. also introduced a change in behaviour. Was that intentional?

Previously, `-certfile` *replaced* any certificates included in the message, now it augments them...

Reply:
> Before this PR, it seems that using -certfile always ignored any certificates in the message, using only -certfile for both the EE cert and chain construction.

@vdukhovni, what you describe here is not the effect of using `-certfile` but of using `-nointern`.
Note the context of the documentation paragraph you are commenting here.

Reply:
After having a close look at the sources and comparing also with CMS, I came to the conclusion that the interpretation I gave yesterday:

> To me, the intended meaning of "The supplied certificates can still be used as untrusted CAs however." is clear: the "supplied certificates" refer to those given with the CLI option (not those in the message).

was not the (originally) intended one. Instead, I am meanwhile sure that the "supplied certificates" were meant to denote the ones included in the message. I will clarify this in the documentation.

Reply:
This clarification is meanwhile done in PR #22285.

Reply:
Moreover, the `-untrusted` option (alias) originally added by this PR is gone after the clarifications achieved meanwhile.
So I believe this conversation thread has been resolved.